packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -550,3 +558,77 @@ class _AdbLogReader extends DeviceLogReader {
     return other.device.id == device.id;
   }
 }
+
+class _AndroidDevicePortForwarder extends DevicePortForwarder {
+  _AndroidDevicePortForwarder(this.device);
+
+  final AndroidDevice device;
+
+  static int _extractPort(String portString) {
+    return int.parse(portString.trim(), onError: (_) => null);
+  }
+
+  Future<List<ForwardedPort>> get forwardedPorts async {
+    final List<ForwardedPort> ports = <ForwardedPort>[];
+
+    String stdout = runCheckedSync(device.adbCommandForDevice(
+      <String>[
+        '-s',
```

Comment:
I just found an issue w/ some of the adb commands we're sending in; `adbCommandForDevice` will pre-pend the device id (`-s device.id`), so we don't need to do it here.